src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/queryAllPackages.hpp

```diff
@@ -0,0 +1,254 @@
+                        fbBuilder,
+                        SyscollectorSynchronization::AttributesUnion::AttributesUnion_syscollector_packages,
+                        syscollectorPackage.Union());
+
+                    auto syncMsg = SyscollectorSynchronization::CreateSyncMsg(
+                        fbBuilder,
+                        agentInfo,
+                        SyscollectorSynchronization::DataUnion::DataUnion_state,
+                        stateMsg.Union());
+
+                    fbBuilder.Finish(syncMsg);
+
+                    std::variant<const SyscollectorDeltas::Delta*,
+                                 const SyscollectorSynchronization::SyncMsg*,
+                                 const nlohmann::json*>
+                        variantData = SyscollectorSynchronization::GetSyncMsg(fbBuilder.GetBufferPointer());
+
+                    // This could throw a exception if there is not information about the os, because the user disable
+                    // it from syscollector.
+                    try
+                    {
+                        auto context = std::make_shared<TScanContext>(variantData);
+                        m_suborchestration->handleRequest(std::move(context));
+                    }
+                    catch (const std::exception& e)
+                    {
+                        logError(WM_VULNSCAN_LOGTAG, "Error handling request: %s", e.what());
+                    }
+                }
+            }
+            else
+            {
+                logWarn(WM_VULNSCAN_LOGTAG,
+                        "Empty response for agent '%s' in Wazuh-DB 'sys_programs' query",
+                        std::string(data->agentId()).c_str());
+            }
+        }
+        else if (data->operationType() == OperationType::InsertAll)
+        {
+            logDebug2(WM_VULNSCAN_LOGTAG, "Re-scanning packages for all agents");
+            for (const auto& agent : data->getMessages())
+            {
+                // Execute query for all packages, modify to adapt to testtool input.
+                std::string agentIdStr;
+                if (agent.at("id").is_string())
+                {
+                    agentIdStr = agent.at("id").template get<std::string>();
+                }
+                else
+                {
+                    agentIdStr = std::to_string(agent.at("id").template get<int>());
+                }
+
+                m_wdbSocketWrapper->query(WazuhDBQueryBuilder::builder().agentGetPackagesCommand(agentIdStr).build(),
+                                          response);
+
+                // Validate the response
+                if (response.empty())
+                {
+                    logWarn(WM_VULNSCAN_LOGTAG,
+                            "Empty response for agent '%s' in Wazuh-DB 'sys_programs' query",
+                            agentIdStr);
+                    continue;
+                }
+
+                for (auto package : response)
```

Comment:
each element copy could be avoided ?
```suggestion
                for (const auto& package : response)
```